alpha/apps/kaltura/modules/extwidget/actions/serveFlavorAction.class.php

```diff
@@ -799,4 +798,84 @@ private function getClipData($path, $flavor)
 		return $clipDesc ;
 	}
 
-}
\ No newline at end of file
+	protected function serveLiveChannel(LiveChannel $entry)
+	{
+		$repeat = $entry->getRepeat();
+		$playlist = entryPeer::retrieveByPK($entry->getPlaylistId());
+
+		list($entryIds, $durations, $referenceEntry, $captionFiles) = myPlaylistUtils::executeStitchedPlaylist($playlist, null);
+
+		// get cycle info
+		$cycleDurations = $durations;
+		$cycleEntryIds = $entryIds;
+		$cycleClipCount = count($cycleDurations);
+		$cycleDuration = array_sum($cycleDurations);
+		$cycleSegmentCount = 0;
+		foreach ($cycleDurations as $duration)
+		{
+			$cycleSegmentCount += ceil($duration / self::LIVE_CHANNEL_SEGMENT_DURATION);
+		}
+
+		// find the start / end time
+		$dvrWindowSize = self::LIVE_CHANNEL_SEGMENT_DURATION * self::LIVE_CHANNEL_SEGMENT_COUNT;
+		$endTime = time() * self::SECOND_IN_MILLISECONDS;
+		$startTime = $endTime - self::TIME_MARGIN - $dvrWindowSize;
+
+		// get the reference time (start time of the first run)
+		$playlistStartTime = $entry->getStartDate('U') * self::SECOND_IN_MILLISECONDS;
+
+		// get the initial cycle info
+		$cycleIndex = floor(($startTime - $playlistStartTime) / $cycleDuration);
+		$clipIndex = $cycleIndex * $cycleClipCount;
+		$segmentIndex = $cycleIndex * $cycleSegmentCount;
+		$durations = array();
+		$entryIds = array();
+
+		$initialSegmentIndex = 0;
+		$initialClipIndex = 0;
+		$firstClipStartTime = 0;
+
+		$currentTime = $playlistStartTime + $cycleIndex * $cycleDuration;
+		while ($currentTime < $endTime)
+		{
+			// get the current clip duration
+			$cycleClipIndex = $clipIndex % $cycleClipCount;
+			$duration = $cycleDurations[$cycleClipIndex];
+			if ($currentTime + $duration > $startTime)
+			{
+				if (!$durations)
+				{
+					// set first clip details
+					$firstClipStartTime = $currentTime;
+
+
+					if (!$repeat)
```

Comment:
Since this var does not really - you can take the value from the entry->repeat propery